### ISSUE TITLE
How to wait for redirects on click?

### ISSUE THREAD

**Issue by overflowz** [segmentio/nightmare#896]
Well, hello. I'm trying to submit form, wait for page load (redirect) and continue work there, but whenever I try to click on element, nightmare does not wait for page reload, but instantly continues queued tasks.

Could someone please add something like `waitForEvent('did-finish-loading')` feature? Or how can I do that?

Thanks!

**Comment by rosshinkley** [comment]
It depends on how the "reload" is happening.  For all actions, the instance should wait for [`did-stop-loading`](https://github.com/segmentio/nightmare/blob/master/lib/runner.js#L153).  I suspect your redirected-to page is finished loading, but perhaps is doing some continuing work asynchronously?  Or, it could be that the redirect is not instant - you _might_ want to add a short wait between form submission and the next task.  Alternatively, you could wait for an element to appear on the page you're redirected to.

**Comment by overflowz** [comment]
#481 is having exactly the same problem. While I tried the code snippet provided there, it does not help much. In examples, there's like goto(..).click(..).wait(..); <-- without this wait, it does not wait until form was submitted and does work instantly. What if I can't find pattern or I have two patterns, like authentication successful or not. What should I wait for? Anyways, I've managed it with nightmare-window-manager. calling waitWindowLoad(..) after click(..) worked fine. It would be nice to have such feature to wait for some specific events happen before executing another task (like the #481 provided). e.g: waitForEvent('dom-ready').then(...).

**Comment by casesandberg** [comment]
You can always wait on a specific div to show up:
```
...
  .click('#formSubmit')
  .wait('#divOnNewPage')
  // continue work
```

I am going to close this, but feel free to re-open if you can't get it work.
